syft/messaging/plan/state.py

```diff
@@ -137,3 +140,62 @@ def detail(worker: AbstractWorker, state_tuple: tuple) -> "State":
 
         state = State(owner=worker, state_placeholders=state_placeholders)
         return state
+
+    @staticmethod
+    def bufferize(worker: AbstractWorker, state: "State") -> StatePB:
+        """
+        Serialize the State to Protobuf message
+        """
+        protobuf_state = StatePB()
+
+        protobuf_placeholders = [
+            sy.serde.protobuf.serde._bufferize(worker, placeholder)
+            for placeholder in state.state_placeholders
+        ]
+        protobuf_state.placeholders.extend(protobuf_placeholders)
+
+        state_tensors = []
+        for tensor in state.tensors():
+            protobuf_tensor = sy.serde.protobuf.serde._bufferize(worker, tensor)
+            state_tensor = StateTensorPB()
```

Comment:
`StateTensorPB` exists to smooth over the multiple possible types in `state.tensors()`, which may include Torch parameters in addition to various other tensor types. It has no corresponding class in Python.

Reply:
@LaRiffle Might be helpful to make a list of what tensor types can be included in the state. I've included Torch tensors and parameters so far, but pretty sure there are more than that, right?